src/ILLink.RoslynAnalyzer/DataFlow/DynamicallyAccessedMembersBinder.cs

```diff
@@ -0,0 +1,435 @@
+						continue;
+
+					// Ignore private properties on a base type - those are completely ignored by reflection
+					// (anything private on the base type is not visible via the derived type)
+					// Note that properties themselves are not actually private, their accessors are
+					if (onBaseType &&
+						(@event.AddMethod == null || @event.AddMethod.DeclaredAccessibility == Accessibility.Private) &&
+						(@event.RemoveMethod == null || @event.RemoveMethod.DeclaredAccessibility == Accessibility.Private))
+						continue;
+
+					if (filter != null && !filter (@event))
+						continue;
+
+					if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Static) {
+						if ((@event.AddMethod != null) && !@event.AddMethod.IsStatic) continue;
+						if ((@event.RemoveMethod != null) && !@event.RemoveMethod.IsStatic) continue;
+					}
+
+					if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Instance) {
+						if ((@event.AddMethod != null) && @event.AddMethod.IsStatic) continue;
+						if ((@event.RemoveMethod != null) && @event.RemoveMethod.IsStatic) continue;
+					}
+
+					if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.Public) {
+						if ((@event.AddMethod == null || (@event.AddMethod.DeclaredAccessibility != Accessibility.Public))
+							&& (@event.RemoveMethod == null || (@event.RemoveMethod.DeclaredAccessibility != Accessibility.Public)))
+							continue;
+					}
+
+					if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.NonPublic) {
+						if ((@event.AddMethod != null) && @event.AddMethod.DeclaredAccessibility == Accessibility.Public) continue;
+						if ((@event.RemoveMethod != null) && @event.RemoveMethod.DeclaredAccessibility == Accessibility.Public) continue;
+					}
+
+					yield return @event;
+				}
+
+				if ((bindingFlags & BindingFlags.DeclaredOnly) == BindingFlags.DeclaredOnly)
+					yield break;
+
+				type = type.BaseType;
+				onBaseType = true;
+			}
+		}
+
+		// declaredOnly will cause this to retrieve interfaces recursively required by the type, but doesn't necessarily
+		// include interfaces required by any base types.
+		public static IEnumerable<ITypeSymbol> GetAllInterfaceImplementations (this ITypeSymbol thisType, OperationAnalysisContext context, bool declaredOnly)
+		{
+			ITypeSymbol? type = thisType;
+			while (type != null) {
+				foreach (var i in type.Interfaces) {
+					yield return i;
+
+					ITypeSymbol? interfaceType = i;
+					if (interfaceType != null) {
+						// declaredOnly here doesn't matter since interfaces don't have base types
+						foreach (var innerInterface in interfaceType.GetAllInterfaceImplementations (context, declaredOnly: true))
+							yield return innerInterface;
+					}
+				}
+
+				if (declaredOnly)
+					yield break;
+
+				type = type.BaseType;
+			}
+		}
+
+		// Issue https://github.com/dotnet/roslyn-analyzers/issues/4845
```

Comment:
I don't think this issue applies here - I believe the warning is asking to pass a symbol equality comparer to the HashSet ctor.

Reply:
I left the suppression since `SymbolEqualityComparer `seems to apply to `ISymbol `but clarified this in the comments